### ISSUE TITLE
missing TCH00(1-3) lint for typing.Callable function signature annotation

### ISSUE THREAD

**Issue by bachmannscode** [astral-sh/ruff#6201]
ruff version: 0.0.280

Due to ```from __future__ import annotations```, the python interpreter can run this snippet without raising a ```NameError```:
```
from __future__ import annotations
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from typing import Callable

def print_hello() -> None:
    print("hello")

def foo() -> Callable[[], None]:
    return print_hello

if __name__ == "__main__":
    foo()()
```
This tells me that ```Callable``` is only used for type checking. Running ```ruff snippet.py --select=TCH --isolated``` does, correctly so, not complain.

If we edit the snippet by removing the type checking guard like so:
```
from __future__ import annotations
from typing import TYPE_CHECKING

from typing import Callable

def print_hello() -> None:
    print("hello")

def foo() -> Callable[[], None]:
    return print_hello

if __name__ == "__main__":
    foo()()
```
Running ```ruff snippet.py --select=TCH --isolated``` still does not complain which imho should with one of "TCH00(1-3)".

Notes:
- When ruff linting the same snippet but instead of importing from ```typing```, one imports from ```collections.abc```, ruff, correctly so, complains with TCH003.
- Ruff also does not complain if the ```__future__``` import is replaced by surrounding the annotation with quotes like so ```"Callable[[], None]"``` which can also be run by the python interpreter without any errors.

**Comment by charliermarsh** [comment]
I believe this is working as intended. The typing module itself is exempted from these rules, since you _need_ to import typing at runtime to get access to TYPE_CHECKING, and so there is no runtime benefit to importing it in a deferred manner.

**Comment by bachmannscode** [comment]
I see, that makes sense. Thanks a lot for the answer!
